railties/lib/rails/generators/rails/app/templates/config/initializers/content_security_policy.rb.tt

```diff
@@ -4,24 +4,23 @@
 # For further information see the following documentation
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 
-# Rails.application.config.content_security_policy do |policy|
-#   policy.default_src :self, :https
-#   policy.font_src    :self, :https, :data
-#   policy.img_src     :self, :https, :data
-#   policy.object_src  :none
-#   policy.script_src  :self, :https
-#   policy.style_src   :self, :https
-#   # Specify URI for violation reports
-#   # policy.report_uri "/csp-violation-report-endpoint"
+# Rails.application.configure do
+#   config.content_security_policy do |policy|
+#     policy.default_src :self, :https
+#     policy.font_src    :self, :https, :data
+#     policy.img_src     :self, :https, :data
+#     policy.object_src  :none
+#     policy.script_src  :self, :https
+#     policy.style_src   :self, :https
+#     # Specify URI for violation reports
+#     # policy.report_uri "/csp-violation-report-endpoint"
+#   end
+#
+#   # Generate session nonces for permitted importmap and inline scripts
+#   config.content_security_policy_nonce_generator = ->(r) { Digest::SHA1.hexdigest(r.session.id.to_s) }
```

Comment:
Seems fine to use the session ID directly, without hashing.

This does hinge on the session ID not being readable (and not guessable) by an attacker trying to XSS. So they sure better be random and not be stored in insecure cookies. Those are the default, but someone with a bespoke setup could get stung.

Reply:
Might adds line’s worth of caution?